src/utils.c

```diff
@@ -184,6 +184,22 @@ int add_iso8601_utc_datetime(char* buf, size_t size) {
 
 #endif
 
+#ifdef ARDUINO
+
+int64_t usec_now()
+{
+	return micros();
+}
+
+void get_time(uint32_t *seconds, uint32_t *micro_seconds)
```

Comment:
The idea here is that different platforms expose a `gettimeofday()` and then the `usec_*()` methods just consume them. Can you change this code to be one of the recognised platforms above?